staging/src/k8s.io/kubelet/config/v1beta1/types.go

```diff
@@ -354,6 +354,12 @@ type KubeletConfiguration struct {
 	// Default: "None"
 	// +optional
 	CPUManagerPolicy string `json:"cpuManagerPolicy,omitempty"`
+	// singleProcessOOMKill, if set, will prevent the `memory.oom.group` flag from being set for container
+	// cgroups in cgroups v2. This causes processes in the container to be OOM killed individually instead of as
+	// a group.
+	// Default: false
+	// +optional
+	SingleProcessOOMKill bool `json:"singleProcessOOMKill,omitempty"`
```

Comment:
is it misleading to accept an explicit `false` on cgroupv1 linux machines?

is it misleading to accept an explicit `true` or `false` on windows machines?

should this be `*bool`, with the following validation behavior:
* on windows, only allow nil
* on cgroupv1 linux, only allow nil and explicit true
* on cgroupv2 linux, allow nil, explicit true and false

and explain what nil does on cgroupv1 and cgroupv2 machines?

Reply:
IIUC we validate against the [internal type](https://github.com/kubernetes/kubernetes/blob/cc27d21ad7b4fe60ee55ccf33fd29a1257e00ada/pkg/kubelet/apis/config/validation/validation.go#L46) today. The internal type uses `bool` instead of `*bool`. 

> on windows, only allow nil

While we can disallow `true`, distinguishing `false` and `nil` requires more change to how we validate. 

@liggitt WDYT? I wonder if we can make the internal type consistent with the `v1beta1` type, i.e. use `*bool`

Reply:
> @liggitt WDYT? I wonder if we can make the internal type consistent with the v1beta1 type, i.e. use *bool

We certainly can, and if we need to distinguish between unset and false, we have to.

Handle the validation combinations above, clearly document what unset means on cgroupv1 and cgroupv2 machines, and carefully handle reading the field to expect a nil value is possible.

Reply:
Thanks. @utam0k Could you update the internal type to also use `*bool`?